test/unit/coroutine/await.cpp

```diff
@@ -0,0 +1,57 @@
+#include <util/time.hpp>
+
+#include <yaclib/async/run.hpp>
+#include <yaclib/coroutine/await.hpp>
+#include <yaclib/coroutine/future_coro_traits.hpp>
+#include <yaclib/executor/thread_pool.hpp>
+#include <yaclib/fault/thread.hpp>
+
+#include <exception>
+#include <iostream>
+#include <stack>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+namespace {
+using namespace std::chrono_literals;
+using namespace yaclib;
+
+TEST(await, just_works_pack) {
+  auto tp = MakeThreadPool();
+  auto coro = [&](IThreadPoolPtr tp) -> Future<int> {
+    auto f1 = yaclib::Run(tp, [] {
+      yaclib_std::this_thread::sleep_for(2ms);
+      return 1;
+    });
+    auto f2 = yaclib::Run(tp, [] {
+      return 2;
+    });
+
+    co_await yaclib::Await(f1, f2);
+    co_return std::move(f1).GetUnsafe().Ok() + std::move(f2).GetUnsafe().Ok();
+  };
+  EXPECT_EQ(coro(tp).Get().Ok(), 3);
```

Comment:
Write something like
1. call coroutine
2. co_await some time in coroutine
3. check that we have a future, that runs immediately after call, but now wait
4. wait
5. check everything fine

```C++
bool kek = 0;
auto was_now = ...
auto future = coro(); // kek = 1 then co_await some sleep time then kek = 2
EXPECT(now - was_now < coro sleep time); 
EXPECT(kek == 1);
future.Get();
EXPECT(now - was_now >= coro sleep)
EXPECT(kek == 2);

```